### PR TITLE
[2.3.3] Generate auth field for dockerCredential and support email field

### PR DESCRIPTION
**Pull request by prachidamle**
https://github.com/rancher/rancher/issues/19177

This PR generates the missing "auth" field for the RegistryCredentials - for both namespaced and project level secrets.

We don't seem to support edit registry secret anymore from UI. So the update would regenerate the auth field only if both username/password are present.

The fix is based on this upstream kubectl fix, which generates this auth field: https://github.com/kubernetes/kubernetes/pull/72344/files#diff-3bf290ed2a11e649da5afa02a5bbfd3a

When we create a dockerCredential secret from Rancher UI, we post the data via K8s REST API - which does not seem to have this "auth" field generation like kubectl does. Hence we will handle the generation of this field while creating the secret.

Update to PR:
Instead of generating "auth" value in rancher api, we will leverage the existing mapper in types repo. Also supporting "email" field of the registryCredential.

Depends on types: https://github.com/rancher/types/pull/972

### CODE REVIEW

**Comment by alena1108** [comment]
@prachidamle can we add an integration test that would create the resource and verify that the field is set

**Comment by prachidamle** [comment]
@alena1108 updated an existing test to check 'auth' field is set

**Comment by prachidamle** [comment]
@alena1108 @kinarashah updates to the PR:

Instead of generating "auth" value in rancher api, we will leverage the existing RegistryCredentialMapper in types repo. Also supporting "email" field of the registryCredential.

Depends on types: https://github.com/rancher/types/pull/972

File: pkg/api/store/secret/store.go [hunk 1]
```diff
@@ -18,10 +18,12 @@ type Store struct {
 }
 
 func (s *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+
```
Comment (alena1108) [inline pkg/api/store/secret/store.go]: nit: doesn't look like new lines are needed here

File: pkg/api/store/secret/store.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ func (s *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
 	t = strings.TrimPrefix(t, "namespaced")
 	t = convert.Uncapitalize(t)
 	data["kind"] = t
+
```
Comment (alena1108) [inline pkg/api/store/secret/store.go]: no need for the new line

**Comment by deniseschannon** [comment]
@prachidamle I've updated this to the `release/v2.3.x` branch - can you rebase it?

**Pull request merged**